pkg/netdevice/netDeviceProvider.go

```diff
@@ -122,6 +122,13 @@ func (np *netDeviceProvider) GetFilteredDevices(devices []types.HostDevice,
 		}
 	}
 
+	// filter by acpiIndexes list
```

Comment:
We need a refactor here. The smallest that comes to my mind is 
a. move each if block to a package private function like:

```
func filterByVendor(nf *types.NetDeviceSelectors, rf types.ResourceFactory, devices []types.HostDevice) []types.HostDevice {
if nf.Vendors != nil && len(nf.Vendors) > 0 {
		if selector, err := rf.GetSelector("vendors", nf.Vendors); err == nil {
			return selector.Filter(filteredDevice)
		}
	}
}
```

b. quick and dirty: remove all `x != nil` as `len(x) == 0`  
https://go.dev/play/p/L7gVCSUcHZu

c. Rethink a different object structure than ResourceFactory and Selector, moving some responsibility. IMO this should be addressed in a different PR

Reply:
Hi @zeeke! Thanks for the review.

I went ahead and implemented your "option B" from above. I also moved the filtering into an abstracted `FilterBySelector`. Let me know if this solution looks ok.

Reply:
Even better than what I was thinking about. Thanks!